EVE.pm

```diff
@@ -0,0 +1,215 @@
+=head1 LICENSE
+
+Copyright [1999-2015] Wellcome Trust Sanger Institute and the EMBL-European Bioinformatics Institute
+Copyright [2016-2022] EMBL-European Bioinformatics Institute
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+=head1 CONTACT
+
+ Ensembl <http://www.ensembl.org/info/about/contact/index.html>
+
+=cut
+
+=head1 NAME
+
+ EVE
+
+=head1 SYNOPSIS
+
+ cp EVE.pm ${HOME}/.vep/Plugins
+ ./vep -i variations.vcf --plugin EVE,file=/path/to/eve/data.vcf.gz
+
+=head1 DESCRIPTION
+
+ This is a plugin for the Ensembl Variant Effect Predictor (VEP) that
+ adds information from EVE (evolutionary model of variant effect).
+ It is only available for GRCh38.
+
+ Please cite EVE publication alongside the VEP if you use this resource:
+ https://www.nature.com/articles/s41586-021-04043-8
+
+###################################################
+# Bash script to merge all VCFs from EVE dataset. #
+###################################################
+
+### BEGIN
+
+# Data link: https://evemodel.org/api/proteins/bulk/download/
+# Input: VCF files by protein (vcf_files_missense_mutations inside zip folder)
+# Output: Compressed Merged VCF file (vcf.gz) + index file (.tbi)
+
+DATA_FOLDER='/<PATH-TO>/vcf_files_missense_mutations' # Fill this line
+OUTPUT_FOLDER='/<PATH-TO>/eve_plugin' # Fill this line
+OUTPUT_NAME='eve_merged.vcf' # Default output name
+
+# Get header from first VCF
+cat `ls ${DATA_FOLDER}/*vcf | head -n1` > header
+
+# Get variants from all VCFs and add to a single-file
+ls ${DATA_FOLDER}/*vcf | while read VCF; do grep -v '^#' ${VCF} >> variants; done
+
+# Merge Header + Variants in a single file
+cat header variants | \
+awk '$1 ~ /^#/ {print $0;next} {print $0 | "sort -k1,1V -k2,2n"}' > ${OUTPUT_FOLDER}/${OUTPUT_NAME};
+
+# Remove temporary files
+rm header variants
+
+# Compress and index
+bgzip ${OUTPUT_FOLDER}/${OUTPUT_NAME};
+tabix ${OUTPUT_FOLDER}/${OUTPUT_NAME}.gz;
+
+### END
+
+=cut
+package EVE;
+
+use strict;
+use warnings;
+
+use Bio::EnsEMBL::Variation::Utils::Sequence qw(get_matched_variant_alleles);
+
+use Bio::EnsEMBL::Variation::Utils::BaseVepTabixPlugin;
+use base qw(Bio::EnsEMBL::Variation::Utils::BaseVepTabixPlugin);
+
+sub new {
+  my $class = shift;
+
+  my $self = $class->SUPER::new(@_);
+
+  $self->expand_left(0);
+  $self->expand_right(0);
+
```

Comment:
You also want to check for assembly because the plugin is specific for only GRCh38 assembly. 
  my $assembly = $self->{config}->{assembly};

  if ($assembly ne "GRCh38") {
    die "Assembly is not GRCh38, EVE only works with GRCh38. \n";
  }

Reply:
Nice touch. This is a important message. Thanks.